samples/DotNet/Microsoft.Azure.EventHubs/Rbac/CustomRole/readme.md

```diff
@@ -1,19 +1,16 @@
 # Receive events with the Event Processor Host in .NET Standard with a custom role which grants Listen claim for Event Hubs and blob claims for Storage accounts.
 
-## Prerequisites
-
-* [Microsoft Visual Studio 2015 or 2017](http://www.visualstudio.com).
-* [.NET Core Visual Studio 2015 or 2017 tools](http://www.microsoft.com/net/core).
-* An Azure subscription.
-* [An event hub namespace and an event hub](event-hubs-quickstart-namespace-portal.md).
-* An Azure Storage account.
 
 ## Run the sample
 
-To run the sample, follow these steps:
+**Note:** This sample uses the legacy Event Hubs library `Microsoft.Azure.EventHubs`. We strongly recommend you to use the current library `Azure.Messaging.EventHubs`. 
+
+See [sample](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/identity/Azure.Identity/samples/DefiningCustomCredentialTypes.md#authenticating-with-the-on-behalf-of-flow) that uses the new library to authenticate after the environment variables for client id and secret are set. 
```

Comment:
What is the reason for us to point to "on behalf flow" here?

Reply:
I would question why we're pointing at the custom credential sample.  Unless I'm missing a nuance, the standard [ClientSecretCredential](https://docs.microsoft.com/dotnet/api/azure.identity.clientsecretcredential) would probably be the more relevant type; the custom role would likely be assigned to a normal service principal.